apps/trading/components/market-state/market-state.tsx

```diff
@@ -0,0 +1,105 @@
+import throttle from 'lodash/throttle';
+import type {
+  MarketData,
+  MarketDataUpdateFieldsFragment,
+  SingleMarketFieldsFragment,
+} from '@vegaprotocol/market-list';
+import { marketDataProvider } from '@vegaprotocol/market-list';
+import { t, useDataProvider } from '@vegaprotocol/react-helpers';
+import { MarketStateMapping, Schema } from '@vegaprotocol/types';
+import { HeaderStat } from '../header';
+import { useCallback, useMemo, useRef, useState } from 'react';
+import * as constants from '../constants';
+
+export const MarketState = ({
+  market,
+}: {
+  market: SingleMarketFieldsFragment | null;
+}) => {
+  const [marketState, setMarketState] = useState<Schema.MarketState | null>(
+    null
+  );
+
+  const throttledSetMarketState = useRef(
+    throttle((state: Schema.MarketState) => {
+      setMarketState(state);
+    }, constants.DEBOUNCE_UPDATE_TIME)
+  ).current;
+
+  const update = useCallback(
+    ({ data: marketData }: { data: MarketData | null }) => {
+      if (marketData) {
+        throttledSetMarketState(marketData.marketState);
+      }
+      return true;
+    },
+    [throttledSetMarketState]
+  );
+
+  const variables = useMemo(
+    () => ({ marketId: market?.id || '' }),
+    [market?.id]
+  );
+  useDataProvider<MarketData, MarketDataUpdateFieldsFragment>({
+    dataProvider: marketDataProvider,
+    update,
```

Comment:
I don't love having to add marketDataProvider everywhere to ensure this stays up to date with the latest state